### PR TITLE
Feature/502

### PR DESCRIPTION
**Pull request by M1ndForce**
#502

### CODE REVIEW

File: src/pages/product-list-page/product-list-filter/price-filter/price-filter.js [hunk 1]
```diff
@@ -1,68 +1,88 @@
...
+      dispatch(
+        setPriceFilter(
+          searchParams
+            .get(priceFilter)
+            .split(',')
+            .map((price) => price * 100)
+        )
+      );
+    } else {
+      // dispatch(
```
Comment (moran711) [inline src/pages/product-list-page/product-list-filter/price-filter/price-filter.js:41]: Please remove this commented line

File: src/pages/product-list-page/product-list-filter/price-filter/price-filter.js [hunk 2]
```diff
@@ -1,68 +1,88 @@
...
-          ...filterData.map(
-            (product) => product.basePrice[currency].value / 100
-          )
-        )}
-        max={Math.max(
-          ...filterData.map(
-            (product) => product.basePrice[currency].value / 100
-          )
-        )}
+        // min={Math.min(
```
Comment (moran711) [inline src/pages/product-list-page/product-list-filter/price-filter/price-filter.js:76]: same

File: src/pages/product-list-page/product-list-filter/product-list-filter.js [hunk 3]
```diff
@@ -3,138 +3,126 @@ import Button from '@material-ui/core/Button';
...
   const dispatch = useDispatch();
-
-  const { filterStatus, filterData, filters, language, currency } = useSelector(
-    ({ Products, Language, Currency }) => ({
-      filterData: Products.filterData,
+  const history = useHistory();
+  const { search } = useLocation();
+  const searchParams = new URLSearchParams(search);
+  const { filters, language, filterData } = useSelector(
+    ({ Products, Language }) => ({
```
Comment (moran711) [inline src/pages/product-list-page/product-list-filter/product-list-filter.js:33]: Please move this selector to sepparate file in `redux `folder

File: src/pages/product-list-page/product-list-filter/product-list-filter.js [hunk 4]
```diff
@@ -3,138 +3,126 @@ import Button from '@material-ui/core/Button';
...
+    history.push(`?${searchParams.toString()}`);
   };
   const filtersOptions = {
     categories: {
       filterName: CATERGORY_TEXT[language].value,
       productFilter: categoryFilter,
-      list: categoriesNames,
-      categories,
-      clearFilter: () => handleFilterClear(setCategoryFilter),
+      list: _.map(
```
Comment (moran711) [inline src/pages/product-list-page/product-list-filter/product-list-filter.js:84]: You can import only map method from lodash like
import {map} from 'lodash'

File: src/pages/product-list-page/product-list-filter/product-list-filter.js [hunk 5]
```diff
@@ -3,138 +3,126 @@ import Button from '@material-ui/core/Button';
...
+        filterData.categories,
+        (category) => category.name[language].value
+      ),
+      categories: filterData.categories,
+      filterAction: setCategoryFilter,
+      labels: 'categoryFilter',
```
Comment (moran711) [inline src/pages/product-list-page/product-list-filter/product-list-filter.js:90]: Please move this to config

File: src/pages/product-list-page/product-list-filter/product-list-filter.js [hunk 6]
```diff
@@ -3,138 +3,126 @@ import Button from '@material-ui/core/Button';
...
+      clearFilter: () => handleFilterClear(setCategoryFilter, 'categoryFilter'),
       filterHandler: (e) =>
-        handleFilterChange(e, setCategoryFilter, categoryFilter, categories)
+        handleFilterChange(e, 'categoryFilter', filterData.categories)
```
Comment (moran711) [inline src/pages/product-list-page/product-list-filter/product-list-filter.js:93]: same

File: src/pages/product-list-page/product-list-filter/product-list-filter.js [hunk 7]
```diff
@@ -3,138 +3,126 @@ import Button from '@material-ui/core/Button';
...
-      filterName: COLORS_TEXT[language].value,
-      productFilter: colorsFilter,
-      list: colorsNames,
-      clearFilter: () => handleFilterClear(setColorsFilter),
-      filterHandler: (e) => handleFilterChange(e, setColorsFilter, colorsFilter)
+      list: _.map(filterData.models, (model) => model.name[language].value),
+      categories: filterData.models,
+      filterAction: setModelsFilter,
+      labels: 'modelsFilter',
+      clearFilter: () => handleFilterClear(setModelsFilter, 'modelsFilter'),
```
Comment (moran711) [inline src/pages/product-list-page/product-list-filter/product-list-filter.js:102]: same

File: src/pages/product-list-page/product-list-filter/product-list-filter.js [hunk 8]
```diff
@@ -3,138 +3,126 @@ import Button from '@material-ui/core/Button';
...
+      filterHandler: (e) =>
+        handleFilterChange(e, 'modelsFilter', filterData.models)
```
Comment (moran711) [inline src/pages/product-list-page/product-list-filter/product-list-filter.js:104]: same

File: src/pages/product-list-page/product-list-filter/product-list-filter.js [hunk 9]
```diff
@@ -3,138 +3,126 @@ import Button from '@material-ui/core/Button';
...
+        filterData.patterns,
+        (pattern) => pattern.name[language].value
+      ),
+      categories: filterData.patterns,
+      filterAction: setPatternsFilter,
+      labels: 'patternsFilter',
+      clearFilter: () => handleFilterClear(setPatternsFilter, 'patternsFilter'),
       filterHandler: (e) =>
-        handleFilterChange(e, setPatternsFilter, patternsFilter)
+        handleFilterChange(e, 'patternsFilter', filterData.patterns)
```
Comment (moran711) [inline src/pages/product-list-page/product-list-filter/product-list-filter.js:118]: same

File: src/pages/product-list-page/product-list-filter/product-list-filter.js [hunk 10]
```diff
@@ -3,138 +3,126 @@ import Button from '@material-ui/core/Button';
...
-    dispatch(
-      setPriceFilter([
-        Math.min(
-          ...filterData.map((product) => product.basePrice[currency].value)
-        ),
-        Math.max(
-          ...filterData.map((product) => product.basePrice[currency].value)
-        )
-      ])
+    const sortQuery = searchParams.get('sort');
```
Comment (moran711) [inline src/pages/product-list-page/product-list-filter/product-list-filter.js:122]: same

File: src/pages/product-list-page/product-list-filter/product-list-filter.js [hunk 11]
```diff
@@ -3,138 +3,126 @@ import Button from '@material-ui/core/Button';
...
+    const quantityPerPage = searchParams.get('countPerPage');
```
Comment (moran711) [inline src/pages/product-list-page/product-list-filter/product-list-filter.js:123]: same

File: src/routes/routes.js [hunk 12]
```diff
@@ -149,6 +129,21 @@ const Routes = () => {
                   );
                 }}
               />
+              {/* <Route */}
```
Comment (moran711) [inline src/routes/routes.js:132]: same

**Review by Haytham1337 — DISMISSED** [review]

**Review by AndriyLymych — APPROVED** [review]

File: src/pages/product-list-page/product-list-filter/product-list-filter.js [hunk 13]
```diff
@@ -3,138 +3,130 @@ import Button from '@material-ui/core/Button';
...
 
-  const handleFilterClear = (setFilter) => {
+  const handleFilterClear = (setFilter, queryName) => {
+    searchParams.set(page, 1);
     dispatch(setFilter([]));
-    dispatch(changeFilterStatus(!filterStatus));
+    searchParams.delete(queryName);
+    history.push(`?${searchParams.toString()}`);
   };
   const filtersOptions = {
```
Comment (moran711) [inline src/pages/product-list-page/product-list-filter/product-list-filter.js:76]: You can move this to custom hook

File: src/pages/product-list-page/product-list-filter/product-list-filter.js [hunk 14]
```diff
@@ -3,138 +3,130 @@ import Button from '@material-ui/core/Button';
...
-    modelNames
-  } = useProductSpecies();
-
-  const handleFilterChange = (
-    { target },
-    setFilter,
-    filter,
-    categoriesList
-  ) => {
+  const handleFilterChange = ({ target }, queryName, categoriesList) => {
```
Comment (moran711) [inline src/pages/product-list-page/product-list-filter/product-list-filter.js:37]: same

File: src/pages/product-list-page/product-list-filter/product-list-filter.js (hunk 13 above)
Comment (M1ndForce) [reply src/pages/product-list-page/product-list-filter/product-list-filter.js]: fixed

File: src/pages/product-list-page/product-list-filter/product-list-filter.js [hunk 15]
```diff
@@ -3,138 +3,130 @@ import Button from '@material-ui/core/Button';
...
+      searchParams.set(queryName, query);
+    } else {
+      searchParams.delete(queryName);
+    }
+    searchParams.set(page, 1);
+    history.push(`?${searchParams.toString()}`);
   };
 
-  const handleFilterClear = (setFilter) => {
+  const handleFilterClear = (setFilter, queryName) => {
```
[the remark this answers is not in the document]
Comment (M1ndForce) [reply src/pages/product-list-page/product-list-filter/product-list-filter.js]: fixed

File: src/pages/product-list-page/product-list-filter/product-list-filter.js (hunk 14 above)
Comment (M1ndForce) [reply src/pages/product-list-page/product-list-filter/product-list-filter.js]: fixed

**Review by moran711 — APPROVED** [review]

**Review by AndriyLymych — APPROVED** [review]

**Review by Haytham1337 — APPROVED** [review]

**Pull request merged**